peak/mapper/binding.py

```diff
@@ -66,9 +79,13 @@ def __init__(self,
         arch_isa : Product,
         ir_isa : Product,
         allow_exists : bool, #allow unbound to be Existential
-        enumeration_scheme : tp.Mapping[type,tp.Callable] = {}
+        custom_enumeration : tp.Mapping[type,tp.Callable] = {}
```

Comment:
Having mutable default argument is recipe for disaster.  The default argument is not constructed each time the function called.

```
def bad(d={}):
  return d
bad()[0] = 0
assert bad() == {0: 0}
```

Reply:
Fixed